File: frontend/src/app/prescricao-medica/medicamento/models/itemPrescricaoMedicamento.ts
import { TipoAprazamento } from './tipoAprazamento';
import { UnidadeDose } from './unidadeDose';
import { UnidadeInfusao } from './unidadeInfusao';
import { Diluente } from './diluente';
import { ViasAdministracao } from './viasAdministracao';
export class ItemPrescricaoMedicamento {
    constructor(
        public id?: number,
        public idMedicamento?: number,
        public idListaMedicamento?: number,
        public dose?: number,
        public frequencia?: number,
        public todasVias?: boolean,
        public bombaInfusao?: boolean,
        public velocidadeInfusao?: number,
        public tempoInfusao?: number,
        public unidadeTempo?: string,
        public inicioAdministracao?: Date,
        public condicaoNecessaria?: boolean,
        public observacaoCondicao?: string,
        public viasAdministracaoId?: number,
        public diluenteId?: number,
        public unidadeInfusaoId?: number,
        public unidadeDoseId?: number,
        public tipoAprazamentoId?: number

    ) {}
}

File: farmacia/src/main/java/br/com/basis/madre/farmacia/domain/evento/ItemPrescricaoMedicamento.java
package br.com.basis.madre.farmacia.domain.evento;

import org.springframework.data.elasticsearch.annotations.Document;
import org.springframework.data.elasticsearch.annotations.Field;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDate;

@Document(indexName = "madre-prescricao-itemprescricaomedicamento")
public class ItemPrescricaoMedicamento implements Serializable {

    private static final long serialVersionUID = 1L;


    private Long id;

    private Long idMedicamento;


    private Long idListaMedicamentos;


    private BigDecimal dose;


    private Integer frequencia;


    private Boolean todasVias;


    private Boolean bombaInfusao;


    private BigDecimal velocidadeInfusao;


    private Integer tempoInfusao;





    private LocalDate inicioAdministracao;


    private Boolean condicaoNecessaria;


    private String observacaoCondicao;



}
